File: frontend-web/e-commerceplatform/src/app/models/product.ts
import { Injectable } from "@angular/core";
import { Category } from "./category";


export class Product {
    id: number;
    name: string;
    description: string;
    price: number;
    category?: Category | null;

    constructor(
        id: number,
        name: string,
        description: string,
        price: number,
        category?: Category | null
    ) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.category = category;
    }
}

File: backend-java/SpringCloudPXL/productcatalog-service/src/main/java/be/pxl/services/domain/Product.java
package be.pxl.services.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Table(name="product")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;
    private String description;
    private Double price;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;
}
